data-prepper-api/src/main/java/com/amazon/dataprepper/model/configuration/PluginSetting.java

```diff
@@ -136,6 +137,88 @@ public String getStringOrDefault(final String attribute, final String defaultVal
         throw new IllegalArgumentException(String.format(UNEXPECTED_ATTRIBUTE_TYPE_MSG, object.getClass(), attribute));
     }
 
+    /**
+     * Returns the value of the specified List<String>, or {@code defaultValue} if this settings contains no value for
+     * the attribute.
+     *
+     * @param attribute    name of the attribute
+     * @param defaultValue default value for the setting
+     * @return the value of the specified attribute, or {@code defaultValue} if this settings contains no value for
+     * the attribute
+     */
+    public List<String> getStringListOrDefault(final String attribute, final List<String> defaultValue) {
+        Object object = getAttributeOrDefault(attribute, defaultValue);
+        if (object == null) {
+            return null;
+        } else if (object instanceof List) {
+            ((List<?>) object).forEach(o -> {
+                if (!(o instanceof String)) {
+                    throw new IllegalArgumentException(String.format(UNEXPECTED_ATTRIBUTE_TYPE_MSG, object.getClass(), attribute));
+                }
+            });
+            return (List<String>) object;
+        }
+        throw new IllegalArgumentException(String.format(UNEXPECTED_ATTRIBUTE_TYPE_MSG, object.getClass(), attribute));
+    }
+
+    /**
+     * Returns the value of the specified Map<String, String> object, or {@code defaultValue} if this settings contains no value for
+     * the attribute.
+     *
+     * @param attribute    name of the attribute
+     * @param defaultValue default value for the setting
+     * @return the value of the specified attribute, or {@code defaultValue} if this settings contains no value for
+     * the attribute
+     */
+    public Map<String, String> getStringMapOrDefault(final String attribute, final Map<String, String> defaultValue) {
+        Object object = getAttributeOrDefault(attribute, defaultValue);
+        if (object == null) {
+            return null;
+        } else if (object instanceof Map) {
```

Comment:
This code can be cleaner by throwing first.

I'd also create a variable just to avoid operating immediately after a cast, but I think that is of less importance.

```
if ( !(object instanceof Map))
  throw new IllegalArgumentException(...)

Map<?, ?> map = object;
map.forEach( ...
```

Reply:
This is definitely cleaner. Just to clarify you are not suggesting that this is done immediately after the getAttributeOrDefault call, but after the if statement for checking if object is null?

Reply:
Since the behavior of a `null` value is to return `null`, then it should be after the null check.

```
Object object = getAttributeOrDefault(attribute, defaultValue);
if (object == null) {
    return null;

if ( !(object instanceof Map))
  throw new IllegalArgumentException(...)

Map<?, ?> map = object;
map.forEach( ...
```